packages/fxa-shared/express/routing.ts

```diff
@@ -105,6 +108,24 @@ export const routing = (app: express.Express, logger: Logger) => {
           method: req.method,
           path: req.url,
         });
+
+        if (err.details && err.details.get('body')) {
+          logger.error('joi.validationError', {
+            err,
+            joiErrors: err.details.get('body')?.message,
+          });
+
+          Sentry.withScope((scope) => {
```

Comment:
Not sure how much you dug into how this works but what you are suggesting isnt possible. The next error handler is actually the celebrate code which ends the request. Since we know this is a validation error from line 105: `if (err && isValidationError(err)) {` the last line of that `if` block (line 130) is this: `validationErrorHandler(err, req, res, next);` which comes from lines 30-35. 

Digging into that bit from celebrate is this
```
exports.errors = (opts = {}) => {
  const finalOpts = { ...internals.DEFAULT_ERRORS_OPTS, ...opts };
  Joi.assert(finalOpts, ERRORSOPTSSCHEMA);

  return (err, req, res, next) => {
  // If this isn't a Celebrate error, send it to the next error handler
    if (!isCelebrateError(err)) {
      return next(err);
    }

    const {
      statusCode,
      message,
    } = finalOpts;

    const validation = {};
    // eslint-disable-next-line no-restricted-syntax
    for (const [segment, joiError] of err.details.entries()) {
      validation[segment] = {
        source: segment,
        keys: joiError.details.map((detail) => EscapeHtml(detail.path.join('.'))),
        message: joiError.message,
      };
    }

    const result = {
      statusCode,
      error: HTTP.STATUS_CODES[statusCode],
      message: message || err.message,
      validation,
    };

    debugger;
    return res.status(statusCode).send(result);
  };
};
```

That last line the code block prevents the Sentry error handler that we have registered with the express server since it will only call `next` (both in our routing middleware and in celebrate) if the error is not a joi validation error.

Ive added some type/casting stuff so that this method will still work with non-celebrate/joi validation errors as you correctly pointed out above but would opt to keep the rest of the solution as is to not get too invasive with the different error handlers (e.g. which ones are called and in which order).

The only other place this is called from is from the content server and I do not imagine the fxa devs would have any issue with this capturing to sentry and logging joi validation issues. Digging into how this all works, it was probably an oversight that they were not already captured since we only recently changed Sentry to only capture 500 errors. Let me know if you have any other concerns.

Reply:
Ah, I misread what `celebrate.errors` does. I thought it always passes on the error to the next error handler.

I agree, I think it's fine to keep as is. Thank you for checking.